src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java

```diff
@@ -388,8 +389,28 @@ public String getGroupName(final byte[] groupId) {
     }
 
     @Override
-    public boolean isRegistered() {
-        return true;
+    public boolean isRegistered(String number) {
+        List<Boolean> result = isRegistered(List.of(number));
+        return result.get(0);
+    }
+
+    @Override
+    public List<Boolean> isRegistered(List<String> numbers) {
+        List<Boolean> results = new ArrayList<Boolean> ();
+        if (numbers.isEmpty()) {
+            return results;
+        }
+        try {
+            Map<String, Pair<String, UUID>> registered;
+            registered = m.areUsersRegistered(new HashSet<String>(numbers));
+            for (String number : numbers) {
```

Comment:
you can move the for loop out of the try/catch block.
And maybe convert it to a stream/map/collect construct.

Reply:
The stream/map/collect construct is no doubt more efficient, but I find it intimidating, like some of the Perl code I encounter from time to time. Please feel free to change what I've submitted to use it.